token/program-2022/src/extension/mod.rs

```diff
@@ -373,6 +373,38 @@ impl<'data, S: BaseState> StateWithExtensionsMut<'data, S> {
         self.init_or_get_extension(true)
     }
 
+    /// Packs the relevant Extension of an ExtensionType into an open slot if not already
+    /// found in the data buffer, otherwise overwrites the Extension
```

Comment:
Currently, `init_extension` doesn't actually overwrite the data part, but maybe it should do a `fill(0)` into the data portion to be safer?
Or even better, what if it initializes it to `Type::default()`?

Reply:
Oh yes, I like the idea of initializing to `Type::default()`

Reply:
Unfortunately, the confidential-transfer types don't implement Default. I'll go with `fill(0)` for now.

Reply:
Ohh, thanks!
That said, I'm going to forgo any changes to that behavior in this PR because I think whether we want to remove [this protection](https://github.com/solana-labs/solana-program-library/blob/3e1dd733968a74c19b7df5b0eda1663da79a2377/token/program-2022/src/extension/mod.rs#L89-:90) is a separate conversation

Reply:
`Zeroable` is very different from `Default`, since a type can implement `Default` with some non-zero values, which may be better in some cases.

The bytemuck docs specify the difference: https://docs.rs/bytemuck/1.7.3/bytemuck/trait.Zeroable.html
"An all-zeroes value may or may not be the same value as the Default value of the type."

It doesn't need to get done in this PR and with these extensions, but there may be an extension that requires default initialization rather than all-zeroes initialization.